### ISSUE TITLE
Create training/validation framework

### ISSUE THREAD

**Issue by jimtjames** [lbussell/cs4641-ml-project#7]
Make the boilerplate code to iterate over the training data, feed inputs into the model, calculate loss, and update the model weights. Repeat this for validation (iterate over the validation set and calculate metrics without updating weights).
